server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -2141,6 +2141,11 @@ IndexMetadata build(boolean repair) {
                 // to be able to build a temporary instance
                 tierPreference = null;
             }
+            if (SearchableSnapshotsSettings.isPartialSearchableSnapshotIndex(settings)
+                && (tierPreference == null || tierPreference.size() != 1 || tierPreference.get(0).equals(DataTier.DATA_FROZEN) == false)) {
+                settings = Settings.builder().put(settings).put(DataTier.TIER_PREFERENCE, DataTier.DATA_FROZEN).build();
+                tierPreference = List.of(DataTier.DATA_FROZEN);
```

Comment:
I think we can make `List.of(DataTier.DATA_FROZEN);` a constant. Not that it matters too much (this is the uncommon path I understand?), but lets deduplicate whatever we can across indices if only out of principle :)